backend/src/services/compilers/AlgoDebugCompilerImpl.ts

```diff
@@ -0,0 +1,43 @@
+import { CompilerRequest, CompilerResponse } from "../../types/compiler";
+import { OutputParser } from "../../utils/OutputParser";
+import { Compiler } from "./compilerFactory";
+
+export class AlgoDebugCompilerImpl implements Compiler {
+    static API_URL: string = "http://192.168.2.132:40042/compile";
+
+    async compile(request: CompilerRequest): Promise<CompilerResponse> {
+        const apiResponse = await fetch(AlgoDebugCompilerImpl.API_URL, {
+            method: "POST",
+            body: JSON.stringify({
+                code: request.code,
+                input: request.input,
+                language: request.language,
+            }),
+            headers: { "Content-Type": "application/json" },
+        });
+        const response: CodexApiResponse = await apiResponse.json();
+
+        if (response.success) {
+            return {
+                success: true,
+                output: new OutputParser(response.output).parse(),
+                errorMessage: null,
+            };
+        } else {
+            return {
+                success: false,
+                output: null,
+                errorMessage: response.errorMessage,
+            };
+        }
+    }
+}
+
+//prettier-ignore
+type CodexApiResponse = {
```

Comment:
zmieniłbym nazwę tego typu albo go zrobił w types z bardziej ogólną nazwą i te trzy properties co ich tu nie ma ale są w CodeX bym dał jako opcjonalne wtedy

Reply:
Nazwy faktycznie zapomniałem zmienić po skopiowaniu z CodeX, więc to poprawię. Ja bym tego typu nie scalał z CodeX bo jakby się coś zmieniało w tych kompilatorach to zmiany byłyby bardziej uciążliwe. Dzięki temu te implementacje są od siebie całkowicie niezależne. Ten typ jest wykorzystywany tylko w tym miejscu, więc też nie widzę potrzeby przenoszenia go do oddzielnego.